src/Analyzers/Core/Analyzers/UseCollectionInitializer/AbstractUseCollectionInitializerDiagnosticAnalyzer.cs

```diff
@@ -239,52 +232,24 @@ private void AnalyzeNode(SyntaxNodeAnalysisContext context, INamedTypeSymbol ien
             ImmutableArray<Location> locations,
             ImmutableDictionary<string, string?>? properties)
         {
-            var syntaxTree = context.Node.SyntaxTree;
             var syntaxFacts = GetSyntaxFacts();
 
-            foreach (var (match, _) in matches)
+            foreach (var match in matches)
             {
-                if (match is TExpressionStatementSyntax)
```

Comment:
this was the code to determine waht to fade.  That logic needs to be used by `use collection initializer` and `use collection expression for builder`.  So it was pulled into a common location.